core/datatrak.c

```diff
@@ -561,11 +578,105 @@ static unsigned char *dt_dive_parser(unsigned char *runner, struct dive *dt_dive
 	free(devdata);
 	return NULL;
 }
+/*
+ * Parses the header of the .add file, returns the number of dives in
+ * the archive (must be the same than number of dives in .log file).
+ */
+static unsigned int wlog_header_parser (struct memblock *mem)
```

Comment:
The function is declared as `unsigned int`, but the result is later assigned to an `int`.

Generally, we should avoid unsigned arithmetic like the plague.